### ISSUE TITLE
Text translator

### ISSUE THREAD

**Issue by nupplaphil** [friendica/friendica#12948]
### Is the feature request related to a problem? Please describe.

A lot of social media platforms are able to translate posts/comments between different languages.

I think this would be a great feature for Friendica as well.

### Describe the feature you'd like

Adding a module "translator" which could be a client for an open source framework. I found something like https://github.com/LibreTranslate/LibreTranslate

I think I would use such a framework as a own service/Docker Container and use the module to communicate. Thus would allow us to share this service between different Instances (like I do have)

### Describe alternatives you've considered

Don't create the add-on :-)

### Additional context

**Comment by MrPetovan** [comment]
Why no addon?

**Comment by nupplaphil** [comment]
it was just a joke, "Describe alternatives" => don't create the addon *g*

**Comment by nupplaphil** [comment]
Yes, I'd see this Feature request as an addon, but only the client part. I won't add a whole translator logic into an addon, seems overkill and not very "service oriented architecture" like :)

**Comment by dogrileycom** [comment]
Mastodon has a hook into DeepL and the free level works fine for small sites, that would be awesome.

**Comment by MrPetovan** [comment]
I agree we shouldn’t have a full translation implementation. We could also have a core stub and different addons to provide the client for various translation services.

**Comment by annando** [comment]
See for example our different default avatar providers.

**Comment by MrPetovan** [comment]
Or the storage backend feature.

**Comment by nupplaphil** [comment]
I'd love to add this feature in the upcoming release after finishing the Database / Addon work :-)
